doc/source/getting_started.rst

```diff
@@ -0,0 +1,113 @@
+and *regions* to be used in a model comparison or scenario exercise following the IAMC
+data format.
+
+A **CodeList** is a list of "allowed terms" (or codes), where each term can have several
+attributes (e.g., description, unit, parent region).
+
+A **RegionProcessor** holds a list of RegionAggregationMappings and a
+DataStructureDefinition. This class is used to facilitate region processing for model
+comparison studies.
+
+A **RegionAggregationMapping** is a mapping that defines on a per-model basis how model
+native regions should be renamed and aggregated to comparison regions.
+
+
+Minimum working example
+-----------------------
+
+This section aims to provide a minimum working example of how to use the nomenclature
+package. It is assumed that the variable templates and model mappings already exist.
+Details on how those are structured and how to start using nomenclature "from scratch"
+can be found here :ref:`usage`. 
+
+The following outlines how to use the nomenclature package:
+
+.. code-block:: python
+
+   # Import the necessary libraries
+   import pyam
+   from nomenclature import DataStructureDefinition, RegionProcessor
+   
+   # Initialize DataStructureDefinition and RegionProcessor giving them the
+   # directories where the codelists and mappings are defined as input.
+   dsd = DataStructureDefinition("definitions/")
+   rp = RegionProcessor.from_directory("mappings/", dsd)
+
+   # Read the data using pyam
+   iam_results_file = "some file"
+   df = pyam.IamDataFrame(iam_results_file)
+
+   # Validate that the data frame only contains allowed regions and variables
+   dsd.validate(df)
+   # Apply region processing to the data
+   df = rp.apply(df)
+
+**Notes**
+
+* The pyam library is required as *DataStructureDefinition.validate()* and
+  *RegionProcessor.apply()* take a *pyam.IamDataFrame* as input.
+
+* *DataStructureDefinition* and *RegionProcessor* are initialized from directories
+  containing yaml files. See :ref:`dir-structure` for details. 
+
+* *DataStructureDefinition.apply()* returns *None* if the data frame only contains   
+  allowed values and raises an error otherwise.
+
+
+.. _dir-structure:
+
+Directory structure for definitions and mappings
+------------------------------------------------
+
+This is the directory structure that needs to be in place in order for the validation and region processing to work:
+
+.. code-block:: bash
+
+   .
+   ├── definitions
+   │   ├── region
+   │   │   ├── ...
+   │   │   └── regions.yaml
+   │   └── variable
+   │       ├── ...
+   │       └── variable.yaml
+   └── mappings
+       ├── model_a.yaml
+       └── ...
+
+**Notes**
+
+* The names of the *definitions* and *mappings* directories are purely conventional and
+  don't carry any special meaning. As long as *DataStructureDefinition* and
+  *RegionProcessor.from_directory()* are pointed to the correct directories everything
+  will work.
```

Comment:
No reason to invite users to start using alternative names...
```suggestion
* The **DataStructureDefinition** will be initialized from the *definitions* folder.
   The **RegionProcessor** will be initialized from the *mappings* folder, if it exists.
```

Reply:
The second part might be a bit misleading as it suggests to me that there is some sort of automated mechanism that checks if a mappings folder exists and only instantiates the region processor if it does.
This is not whats currently implemented. If you run:

```python
dsd = DataStructureDefinition("definitions")
rp = RegionProcessor("mappings", dsd)
```
without a directory called "mappings" you will get an error.
So if we want to advertise the fact that "mappings" is optional we should also make it clear that if there are no mappings, `RegionProcessor` must not be used, otherwise the program will crash.

Reply:
Ok, how about "The optional folder *mappings* can be used to initialize a **RegionProcessor**.

Reply:
I wrote an explanation for the next update.